### PR TITLE
Improve `kyuubi-defaults.conf.template`

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

This PR changes the `kyuubi-defaults.conf.template` because

- remove deprecated conf `kyuubi.frontend.bind.port`
- add some properties which were frequently asked by users

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

image

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by turboFei — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

File: conf/kyuubi-defaults.conf.template [hunk 1]
```diff
@@ -17,10 +17,19 @@
...
 
+# Details in https://kyuubi.readthedocs.io/en/master/deployment/settings.html
+
 #
-# kyuubi.authentication           NONE
-# kyuubi.frontend.bind.host       localhost
-# kyuubi.frontend.bind.port       10009
+# kyuubi.authentication                    NONE
+#
+# kyuubi.frontend.bind.host                <undefined>
```
Comment (yaooqinn) [inline conf/kyuubi-defaults.conf.template:25]: why use `<undefined>`?

Comment (pan3793) [reply conf/kyuubi-defaults.conf.template:25]: the default value is undefined, and Kyuubi will bind the first non-localhost network card.

**Comment by yaooqinn** [comment]
> all properties should be the default value to make it intuitive for users

I do not get it

**Comment by pan3793** [comment]
```
kyuubi.frontend.bind.host localhost
```
This is an example, the user may think Kyuubi binds `localhost` in default when looking `kyuubi-defaults.conf.template`, but actually not.

File: docs/deployment/settings.md [hunk 2]
```diff
@@ -122,9 +122,18 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
 ## Kyuubi Configurations
 
 #
-# kyuubi.authentication           NONE
```
Comment (yaooqinn) [inline docs/deployment/settings.md:125]: We may give a much more practical example.

The current example is from the template directly, where all examples for individual configs are commented out, while we shall give meaningful ones on the contrary.

Or, for legible, we could delete this entire block here. For example, `You can configure the Kyuubi properties in $KYUUBI_HOME/conf/kyuubi-defaults.conf, see $KYUUBI_HOME/conf/kyuubi-defaults.conf.template as an example,. `

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Pull request closed without merging**